unity/Assets/Scripts/ArtGallery/Util/NaiveLighthouseToLighthouseVisibility.cs

```diff
@@ -166,12 +175,15 @@ public static class NaiveLighthouseToLighthouseVisibility
                 // If the vertex cannot be seen by one other vertex return 
                 if (!VisibleToOtherVertex(vertex, otherVertexes, polygon))
                 {
-                    return false;
+                    Debug.Log(vertex + " not guarded");
+                    var unguardedIndicator = Resources.Load("Prefabs/NotGuarded") as GameObject;
+                    GameObject.Instantiate(unguardedIndicator, new Vector3(vertex.x, vertex.y, -5f), Quaternion.identity);
+                    allGuarded = false;
                 }
             }
 
             // if every vertex can be seen by one other vertex return true
-            return true;
+            return allGuarded;
```

Comment:
Instead of destroying and creating the game objects here, I would move it to the controller. You can perhaps copy this method there so it also works with the new smart one that will be added, but I would not create the game objects etc. in these scripts.

I would also suggest not creating the game objects at every check, but instead instantiate them once for every light house at creation of the light house in the controller. Then you can add the prefab as a child of the light house game object so you don't need to modify the position and you don't have this problem:

![2020-01-18_14-57-29](https://user-images.githubusercontent.com/1112623/72664926-b25d6700-3a03-11ea-9d1e-f86b69ecb28a.gif)

Then, instead of creating and removing them I would just enable/disable the game objects which should be much more performant. I would also suggest making sure that you don't load the prefab manually using `Resources.Load` but add a new property to the guarded vertex controller just like `m_lighthousePrefab` so you can just drag the prefab there.

Reply:
Partly same as below but yes I agree good ideas, this would possibly be a solution to below as well so moving that method and having the sprites paired already at instantiation would make it better indeed. I'll look into this.

Reply:
Did you already solve this? if not, there should be a method to in the guard to guard interface to get for each lighthouse  the lighthouses it can see, 
You can use this to find out in the controler whether things are unguarded